File: C++/Cristian Maldonado/Grafos/Lista_De_Adyacencia.cpp
#include<bits/stdc++.h>
#include<cstdlib>
#define MAX_V 101

using namespace std;

struct Edge{
	int to = 0, w = 0;
	Edge(int _to, int _w){
		to = _to;
		w = _w;
	}
};

struct Vertex{
	list <Edge> adj;
};

static Vertex G[MAX_V];

int main (int argc, char *argv[]) {
	int V, E;
	scanf("%i %i", &V, &E);
	for(int i = 0; i < V; i++){
		G[i] = Vertex();
	}
	for(int i = 0; i < E; i++){
		int desde, hasta, peso;
		scanf("%i %i %i", &desde, &hasta, &peso);
		G[desde].adj.push_back(Edge(hasta, peso));
		G[hasta].adj.push_back(Edge(desde, peso));
	}
	for(int i = 0; i < V; i++){
		cout << i <<"-> ";
		for(Edge e:G[i].adj){
			cout <<e.to<<" "<<e.w<<"|";
		}
		cout<<""<<endl;
	}
	return 0;
}

File: Codigos estudiantes por lenguaje/Java/Cristian Maldonado/Grafos/Lista_De_Adyacencia.java
import java.util.ArrayList;
import java.util.Scanner;
import java.util.List;

public class Lista_De_Adyacencia {

    static Vertex[] G;
    static Scanner sc = new Scanner(System.in);

    public static void main(String[] args) {
        int V, E;
        V = sc.nextInt();
        E = sc.nextInt();
        G = new Vertex[V];
        for (int i = 0; i < V; i++) {
            G[i] = new Vertex();
            G[i].adj = new ArrayList<>();
        }
        for (int i = 0; i < E; i++) {
            int desde, hasta, peso;
            desde = sc.nextInt();
            hasta = sc.nextInt();
            peso = sc.nextInt();
            G[desde].adj.add(new Edge(hasta, peso));
            G[hasta].adj.add(new Edge(desde, peso));
        }
        for (int i = 0; i < G.length; i++) {
            System.out.print(i + "-> ");
            for (Edge e : G[i].adj) {
                System.out.print(e.to + " " + e.w + "|");
            }
            System.out.println();
        }
    }

    static class Vertex {

        List<Edge> adj;

        public Vertex() {
            adj = new ArrayList<>();
        }
    }

    static class Edge {

        int to, w;

        public Edge(int to, int w) {
            this.to = to;
            this.w = w;
        }
    }
}
